pkg/internal/apis/config/validate.go

```diff
@@ -113,14 +115,22 @@ func (n *Node) Validate() error {
 		errs = append(errs, errors.New("image is a required field"))
 	}
 
+	portMappingsSet := sets.NewString()
+	addrSet := sets.NewString()
+
 	// validate extra port forwards
 	for _, mapping := range n.ExtraPortMappings {
 		if err := validatePort(mapping.HostPort); err != nil {
 			errs = append(errs, errors.Wrapf(err, "invalid hostPort"))
 		}
+
 		if err := validatePort(mapping.ContainerPort); err != nil {
 			errs = append(errs, errors.Wrapf(err, "invalid containerPort"))
 		}
+
+		if err := validatePortMapping(portMappingsSet, addrSet, mapping); err != nil {
```

Comment:
ok, I think you got the mechanics but this implementation inside this loop is confusing because we are really cross-validating between the different portmaps, and this loop is aggregating those inside that is not obvious

You should make  `validatePortMapping` to accept the array of `n.ExtraPortMappings ` directly, and create the sets and all inside it, moving the function outside of this loop